packages/block-library/src/list-item/hooks/use-outdent-list-item.js

```diff
@@ -29,93 +30,90 @@ export default function useOutdentListItem( clientId ) {
+		return parentListItem;
+	}
+
 	return [
 		canOutdent,
-		useCallback( () => {
-			const _hasMultiSelection = hasMultiSelection();
-			const clientIds = _hasMultiSelection
-				? getMultiSelectedBlockClientIds()
-				: [ clientId ];
+		useCallback( ( clientIds = getSelectedBlockClientIds() ) => {
+			clientIds = castArray( clientIds );
 
-			const selectionStart = getSelectionStart();
-			const selectionEnd = getSelectionEnd();
+			if ( ! clientIds.length ) return;
 
-			const listParentId = getBlockRootClientId( clientId );
-			const listAttributes = getBlockAttributes( listParentId );
-			const listItemParentId = getBlockRootClientId( listParentId );
-			const listItemParentAttributes =
-				getBlockAttributes( listItemParentId );
+			const firstClientId = first( clientIds );
 
-			const firstIndex = getBlockIndex( first( clientIds ) );
-			const lastIndex = getBlockIndex( last( clientIds ) );
-			const siblingBlocks = getBlock( listParentId ).innerBlocks;
-			const previousSiblings = siblingBlocks.slice( 0, firstIndex );
-			const afterSiblings = siblingBlocks.slice( lastIndex + 1 );
+			// Can't outdent if it's not a list item.
+			if ( getBlockName( firstClientId ) !== listItemName ) return;
 
-			// Create a new parent list item block with just the siblings
-			// that existed before the first child item being outdent.
-			const newListItemParent = createListItem(
-				listItemParentAttributes,
-				listAttributes,
-				previousSiblings
-			);
+			const parentListItem = getParentListItem( firstClientId );
 
-			const lastBlock = getBlock( last( clientIds ) );
-			const childList = lastBlock.innerBlocks[ 0 ];
-			const childItems = childList?.innerBlocks || [];
-			const hasChildItems = !! childItems.length;
+			// Can't outdent if it's at the top level.
+			if ( ! parentListItem ) return;
 
-			const newBlocksExcludingLast = clientIds
-				.slice( 0, -1 )
-				.map( ( _clientId ) => cloneBlock( getBlock( _clientId ) ) );
-
-			// Create a new list item block whose attributes are equal to the
-			// last block being outdent and whose children are the children that it had (if any)
-			// followed by the siblings that existed after it.
-			const newLastItem = createListItem(
-				lastBlock.attributes,
-				hasChildItems ? childList.attributes : listAttributes,
-				[ ...childItems, ...afterSiblings ]
+			const parentList = getBlockRootClientId( firstClientId );
+			const lastClientId = last( clientIds );
+			const order = getBlockOrder( parentList );
+			const followingListItems = order.slice(
+				getBlockIndex( lastClientId ) + 1
 			);
 
-			// Replace the parent list item block, with a new block containing
-			// the previous siblings before the first block being outdent,
-			// followed by the blocks being outdent with the after siblings added
-			// as children of the last block.
-			replaceBlocks(
-				[ listItemParentId ],
-				[ newListItemParent, ...newBlocksExcludingLast, newLastItem ]
-			);
```

Comment:
@jorgefilipecosta @ntsekouras I cleaned up outdenting to use `moveBlocksToPosition`. This has the benefit of keeping block client IDs and having to reapply selection. It's also a lot simpler and easier to follow.